gems/pending/appliance_console/database_replication.rb

```diff
@@ -0,0 +1,93 @@
+require 'appliance_console/logging'
+require 'appliance_console/prompts'
+require 'English'
+
+module ApplianceConsole
+  class DatabaseReplication < DatabaseConfiguration
+    include ApplianceConsole::Logging
+
+    REPMGR_CONFIG = '/etc/repmgr.conf'.freeze
+
+    attr_accessor :cluster_name, :node_number
+
+    def initialize(hash = {})
+      super
+    end
+
+    def ask_for_cluster_name
+      self.cluster_name = just_ask("cluster name", cluster_name)
+    end
+
+    def ask_for_node_number
+      self.node_number = ask_for_integer("node number")
+    end
+
+    def confirm
+      clear_screen
+      say(<<-EOL)
+Replication Server Configuration
+
+        Cluster Name: #{cluster_name}
+        Node Number:  #{node_number}
+        Host:         #{host}
+        Username      #{username}
+        Database      #{database}
+        Password      #{password}
+        EOL
+
+      agree("Apply this Replication Master Configuration? (Y/N): ")
+    end
+
+    def already_configured?
+      if File.exist?(REPMGR_CONFIG)
+        say("Warning: File #{REPMGR_CONFIG} exists. Replication is already configured")
+        Logging.logger.info("Warning: File #{REPMGR_CONFIG} exists. Replication is already configured")
+        return agree("Continue with configuration? (Y/N): ")
+      end
+      true
+    end
+
+    def create_config_file(node)
```

Comment:
@carbonin Agreed the host naming is confusing. Implementing your suggestion of `ask_for_database_credentials should be reimplemented for these classes` and not pulling it from database_configuration will resolve this and many of your other comments.